lib/lumberjack/beats/server.rb

```diff
@@ -66,8 +66,16 @@ def run(&block)
         # unless the server is closing
         next unless connection
 
+        Thread.abort_on_exception = true
         Thread.new(connection) do |connection|
-          connection.run(&block)
+          begin
+            connection.run(&block)
+          rescue Lumberjack::Beats::Connection::ConnectionClosed 
+            # Connection will raise a wrapped exception upstream, 
+            # but if the threads are managed by the library we can simply ignore it.
```

Comment:
The ruby-lumberjack has 2 way to work, which is a bit unfriendly:

-  it can manage the thread for you but does it in a really minimal way
- or you can do all the managing yourself (which is what the input is doing) 

But for integration testing for the library its handy to have a way to let it do all the heavy lifting